### PR TITLE
Injective watcher: change to explorer rpc

### PR DESCRIPTION
**Pull request by panoel**
This uses the Injective explorer/api endpoint

### CODE REVIEW

File: watcher/src/watchers/CosmwasmWatcher.ts [hunk 1]
```diff
@@ -138,6 +170,74 @@ export class CosmwasmWatcher extends Watcher {
...
+          const numEvents = hashResult.data.messages.length;
+          for (let j = 0; j < numEvents; j++) {
+            let type: string = hashResult.data.messages[j].type;
+            if (type.includes('cosmwasm.wasm')) {
+              let contract = hashResult.data.messages[j].value.contract;
+              if (contract !== address) {
+                // Not on the core contract
+                continue;
+              }
+              let emitter = hashResult.data.messages[j].value.sender;
```
Comment (evan-gray) [inline watcher/src/watchers/CosmwasmWatcher.ts:220]: I think this sender is the wallet, but we want the emitter address

File: watcher/src/watchers/CosmwasmWatcher.ts [hunk 2]
```diff
@@ -138,6 +170,74 @@ export class CosmwasmWatcher extends Watcher {
...
+              let sequence = hashResult.data.signatures[j].sequence;
```
Comment (evan-gray) [inline watcher/src/watchers/CosmwasmWatcher.ts:221]: I think this sequence is the wallet sequence, but we want the wormhole core log sequence number

**Review by evan-gray — CHANGES_REQUESTED** [review]

**Comment by evan-gray** [comment]
Confirmed the explorer endpoints do not have logs available at this time so this doesn't look feasible as there's no way to determine which transactions involve the core bridge.

File: watcher/src/watchers/CosmwasmWatcher.ts (hunk 1 above)
Comment (panoel) [reply watcher/src/watchers/CosmwasmWatcher.ts]: Fixed

File: watcher/src/watchers/CosmwasmWatcher.ts (hunk 2 above)
Comment (panoel) [reply watcher/src/watchers/CosmwasmWatcher.ts]: Fixed

**Review by evan-gray — CHANGES_REQUESTED** [review]
I think there's a faster way to do `getMessagesForBlocks_explorer`, if you simply call `api/explorer/v1/txs` and then paginate back until getting a transaction which the block is `< fromBlock`. Also wondering if this should just be moved to it's own class - might be cleaner :)

**Comment by panoel** [comment]
Moved Injective to separate class.

**Comment by evan-gray** [comment]
https://github.com/wormhole-foundation/wormhole-dashboard/pull/62

**Pull request closed without merging**